### PR TITLE
doc: update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by miekg**
Remove bunch of old/stale text, but add a section (and clarify) using
issues to describe a future PR and that those need to be small (if you
want it merged faster)

### CODE REVIEW

**Comment by stp-ip** [comment]
looks good. Waiting for another lgtm.

**Review by fturib — COMMENTED** [review]
I am ok with those changes.

I think the part "New features" should be more detailed. 
But that can be a second step for this doc

Should I open an other PR to have conversation on how we could improve the "new feature" workflow ?

Thanks!

**Comment by fturib** [comment]
(argh.. I lost the comment I did on the new features... rewriting below)

About **new feature**

I feel that it would help a lot if we could have a conversation and approval early on in the development of the new features.

Would it help, if we define a workflow with PR on the design and implementation overview ?
We could: 
- have feedback on target of the feature
- get advices on other way to reach it 
- ensure we agree on the way to implement (plugin, external plugin, enhancement, libs, etc ..)
- propose steps - I mean how to split implementation into several simple PR (more easy to review)

Conversation can go on until that PR is approved.

I guess once approved, it would be more easy to do the review, and focus on each part of code.
And because the design and implementation direction is approved, maybe several maintainers can review the code.

As said in precedent comment, should I open a PR to propose inclusion of such process in this `contributing` doc  ?

**Comment by chrisohaver** [comment]
> , if we define a workflow with PR on the design and implementation overview

A PR to... i guess a new repo?  e.g. `coredns/feature`

**Comment by miekg** [comment]
depends on how big a feature is; some require a complete design doc and can't even be (fully) described in a issue.  I.e. the cache rewrite is almost big enough to actually have a design doc behind it.

Apparently noone read this CONTRIBUTING.md before, so I'm not in favor of adding a whole bunch of new text (although moving it to .github might help to make it more visible).

File: CONTRIBUTING.md
Comment (chrisohaver) [inline CONTRIBUTING.md]:
```suggestion
feature. Smaller means: easier to understand and review. This in turn means things can be merged
```

**Review by chrisohaver — COMMENTED** [review]
LGTM, one minor grammar nit.

**Comment by yongtang** [comment]
LGTM

**Comment by fturib** [comment]
LGTM

**Review by chrisohaver — APPROVED** [review]
LGTM

**Pull request merged**
